ffs/grpc/service.go

```diff
@@ -332,31 +332,44 @@ func (s *Service) PushConfig(ctx context.Context, req *pb.PushConfigRequest) (*p
 		return nil, err
 	}
 
-	config := ffs.CidConfig{
-		Cid: c,
-		Hot: ffs.HotConfig{
-			Enabled:       req.Config.Hot.Enabled,
-			AllowUnfreeze: req.Config.Hot.AllowUnfreeze,
-			Ipfs: ffs.IpfsConfig{
-				AddTimeout: int(req.Config.Hot.Ipfs.AddTimeout),
+	options := []api.PushConfigOption{}
+
+	if req.HasConfig {
```

Comment:
Here's where we use our specially encoded information about optionally set options.